### PR TITLE
feat: add h3 to benchmarks

### PR DESCRIPTION
**Pull request by pi0**
Add [unjs/h3](https://github.com/unjs/h3) to benchmarks.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: benchmarks/h3.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+'use strict'
+
+const { createServer } = require('http')
+const { createApp } = require('h3')
+
+const app = createApp()
+
+app.useAsync('/', () => ({ hello: 'world' }))
```
Comment (pi0) [inline benchmarks/h3.js:8]: The router is fast skipped for this case. Is it still eligible to set `hasRouter` flag or are there any baselines a router should handle?

Comment (RafaelGSS) [reply benchmarks/h3.js:8]: cc: @mcollina

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Comment by manniL** [comment]
@RafaelGSS Is there anything blocking the merge so far? ☺️

**Comment by RafaelGSS** [comment]
well... that's too old. it needs a rebase for sure

**Comment by manniL** [comment]
Created an updated version of this PR in #258 ☺️

**Pull request closed without merging**